gemstone/src/swapper/chainflip/provider.rs

```diff
@@ -0,0 +1,177 @@
+    fn map_asset_id(asset: &QuoteAsset) -> ChainflipAsset {
+        let asset_id = asset.asset_id();
+        let chain_name = capitalize_first_letter(asset_id.chain.as_ref());
+        ChainflipAsset {
+            chain: chain_name,
+            asset: asset.symbol.clone(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Swapper for ChainflipProvider {
+    fn provider(&self) -> &SwapProviderType {
+        &self.provider
+    }
+
+    fn supported_assets(&self) -> Vec<SwapChainAsset> {
+        vec![
+            SwapChainAsset::Assets(Chain::Bitcoin, vec![]),
+            SwapChainAsset::Assets(Chain::Ethereum, vec![ETHEREUM_USDC.id.clone(), ETHEREUM_USDT.id.clone()]),
+            SwapChainAsset::Assets(Chain::Solana, vec![SOLANA_USDC.id.clone()]),
+            SwapChainAsset::Assets(Chain::Arbitrum, vec![ARBITRUM_USDC.id.clone()]),
+        ]
+    }
+
+    async fn fetch_quote(&self, request: &SwapQuoteRequest, provider: Arc<dyn AlienProvider>) -> Result<SwapQuote, SwapperError> {
+        let chainflip_client = ChainflipClient::new(provider.clone());
+        let broker_client = BrokerClient::new(provider);
+        let src_asset = Self::map_asset_id(&request.from_asset);
+        let dest_asset = Self::map_asset_id(&request.to_asset);
+        let fee_bps = get_swap_config().referral_fee.thorchain.bps;
+        let amount = request.value.parse::<U256>()?;
+
+        let quote_request = QuoteRequest {
+            amount: request.value.clone(),
+            src_chain: src_asset.chain.clone(),
+            src_asset: src_asset.asset.clone(),
+            dest_chain: dest_asset.chain,
+            dest_asset: dest_asset.asset,
+            is_vault_swap: true,
+            dca_enabled: true,
+            broker_commission_bps: Some(fee_bps),
+        };
+
+        let swap_limit_req = broker_client.get_swap_limits();
+        let quote_req = chainflip_client.get_quote(&quote_request);
+        let (swap_limit, quote_responses) = futures::try_join!(swap_limit_req, quote_req)?;
+
+        if swap_limit.get_min_deposit_amount(&src_asset)? > amount {
+            return Err(SwapperError::InputAmountTooSmall);
+        }
+
+        if quote_responses.is_empty() {
+            return Err(SwapperError::NoQuoteAvailable);
+        }
+
+        let quote_response = &quote_responses[0];
+        let quote = SwapQuote {
+            from_value: request.value.clone(),
+            to_value: quote_response.egress_amount.clone(),
+            data: SwapProviderData {
+                provider: self.provider.clone(),
+                slippage_bps: quote_response.slippage_bps(),
+                routes: vec![SwapRoute {
+                    input: request.from_asset.asset_id(),
+                    output: request.to_asset.asset_id(),
+                    route_data: "{}".to_string(),
+                    gas_limit: None,
+                }],
+            },
+            eta_in_seconds: Some(quote_response.estimated_duration_seconds as u32),
```

Comment:
This is one that triggers a lot of debate, some want the eta to only be the inbound + swap time, some want inbound + swap + egress, just mentioning to think about, since for example with BTC as soon as it is in the mempool after a swap, the user can spend it, without having to wait for 3 confirmations (which is included in estimated_duration_seconds)

Reply:
yeah, we will show total time